src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1447 @@
+
+        loss = None
+        if labels is not None:
+            if self.config.problem_type is None:
+                if self.num_labels == 1:
+                    self.config.problem_type = "regression"
+                elif self.num_labels > 1 and (labels.dtype == torch.long or labels.dtype == torch.int):
+                    self.config.problem_type = "single_label_classification"
+                else:
+                    self.config.problem_type = "multi_label_classification"
+
+            if self.config.problem_type == "regression":
+                loss_fct = MSELoss()
+                if self.num_labels == 1:
+                    loss = loss_fct(logits.squeeze(), labels.squeeze())
+                else:
+                    loss = loss_fct(logits, labels)
+            elif self.config.problem_type == "single_label_classification":
+                loss_fct = CrossEntropyLoss()
+                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
+            elif self.config.problem_type == "multi_label_classification":
+                loss_fct = BCEWithLogitsLoss()
+                loss = loss_fct(logits, labels)
+
+        if not return_dict:
+            output = (logits,)
+            output = (output + (outputs[1:],)) if output_hidden_states else output
+            return ((loss,) + output) if loss is not None else output
+
+        return ImageClassifierOutput(
+            loss=loss,
+            logits=logits,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+
+@add_start_docstrings(
+    """Beit3ForCaptioning has a linear head on top of Beit3Model for image captioning. BEiT-3 is a multimodal
+    foundation model, The key idea in BEiT-3 is to model images as another language. Beit3 uses multiway Transformers
+    architecture which uses a shared self-attention module.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3ForCaptioning(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super(Beit3ForCaptioning, self).__init__(config)
+
+        self.beit3 = Beit3Model(config)
+        self.mlm_classifier = nn.Linear(config.hidden_size, config.vocab_size)
+
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_CAPTIONING_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=CausalLMOutputWithPast, config_class=_CONFIG_FOR_DOC)
+    def forward(
+            self,
+            input_ids,
+            pixel_values,
+            attention_mask,
+            language_masked_pos=None,
+            text_len=None,
+            past_key_value=None,
+            output_hidden_states: Optional[bool] = None,
+            output_attentions: Optional[bool] = None,
+            return_dict: Optional[bool] = None,
+            labels: Optional[torch.LongTensor] = None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3ForCaptioning, Beit3Processor
```

Comment:
This code snippet does not show how to do captioning, would be great to clarify that.

Also, no need to do torch.tensor, please specify return_tensors="pt" to the processor to create PyTorch tensors

Reply:
The way the task is setup in beit3 is , they will fill the word in places where there is spl token (64001) in input_ids. So it is not actually captioning in usual sence.

Reply:
This snippet should demonstrate that then

Reply:
The current examples just does it, I have added comments to explain this.

Reply:
Let me know if I need to add anything more. (would be great if you can point me good examples)